### ISSUE TITLE
Courses menu stops functioning when having many courses

### ISSUE THREAD

**Issue by cdipe** [gjb2048/moodle-theme_essential#849]
When having the Courses menu set to "Last accessed time or enrolment start time if never accessed", the menu stops functioning as expected when having a lot of courses. 

It still shows your courses, but doesn't update the list and last visited course may not show at all in the menu (depending on how many it is set to show).

Our test person had 140+ courses, and the menu started acting correctly when we removed some of them. Unfortunately I don't have an exact number.

Using Moodle 3.2.5 and Essential 3.2.1.3

**Comment by gjb2048** [comment]
Hi Peter,

Ok, given that, what is wrong with the code: https://github.com/gjb2048/moodle-theme_essential/blob/MOODLE_32/classes/output/core_renderer.php#L711-L772 please?

So you're saying that the menu was broken with 140 courses?  Could you test this again to verify please?

How can I replicate the problem please?

When you say 'does not update', how do you know this?  What steps were taken to get to that conclusion.  What evidence is there?

Are there any out of memory errors?

Without an exact replicate-able number then this is speculation and not fact.

G

**Comment by cdipe** [comment]
Hi Gareth,
sorry about the faulty description. We used "login as" when trying to solve the problem for the user who had this problem, and removed a lot of courses for the user before remembering that the Courses menu doesn't update when using "login as" (which is a good thing, but not in this case). 
So, when we asked the user with the problem to login again, the problem was solved without us having a specific number of courses, or a specific course, that was removed and solved the problem.
I have now added 150 courses to myself, without being able to replicate the problem... Must have been an odd incident, and not depending specifically on the number of courses. If the problem occurs again we'll keep a closer look when trying to solve it.
